### PR TITLE
Refactor get relative path of a give file to a util function

### LINKED ISSUE #32: Get relative path of a file can be refactored in to a util function
**Description:**
The logic used to get relative path of a file in `create.go` [segment1](https://github.com/wso2/update-creator-tool/blob/master/cmd/create.go#L802-L807) and in `validate.go` [segment2](https://github.com/wso2/update-creator-tool/blob/master/cmd/validate.go#L376-L381) can be refactored into a util function and used in both commands

**Suggested Labels:**
Improvement

**Suggested Assignees:**
Kasun Siyambalapitiya

**Affected Product Version:**
N/A
**OS, DB, other environment details and versions:**    
N/A
**Steps to reproduce:**
N/A

**Related Issues:**
N/A

### PR DESCRIPTION
**Pull request by kasunsiyambalapitiya**
## Purpose
Resolves #32 

## Goals
Refactor the logic used to get the relative path of a given file to a util function and use it in both `create.go` and `validate.go`

## Approach
N/A

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
 - Unit tests 
N/A
 - Integration tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? N/A for Go
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A
 
## Learning
N/A.

### CODE REVIEW

File: cmd/create.go [hunk 1]
```diff
@@ -793,16 +793,7 @@ func readZip(location string) (node, error) {
...
-		if strings.Contains(file.Name, "/") {
-			relativePath = strings.SplitN(file.Name, "/", 2)[1]
-		} else {
-			relativePath = file.Name
-		}
-
-		// Replace all \ with /. Otherwise it will cause issues in Windows OS.
-		relativePath = filepath.ToSlash(relativePath)
-		logger.Trace(fmt.Sprintf("relativePath: %s", relativePath))
+		var relativePath string = util.GetRelativePath(file)
```
Comment (nishadi) [inline cmd/create.go]: We can use the := short assignment statement in this var declaration implicitly.

File: cmd/validate.go [hunk 2]
```diff
@@ -367,12 +367,7 @@ func readDistributionZip(filename string) (map[string]bool, error) {
 	for _, file := range zipReader.Reader.File {
 		logger.Trace(file.Name)
 
-		var relativePath string
-		if strings.Contains(file.Name, "/") {
-			relativePath = strings.SplitN(file.Name, "/", 2)[1]
-		} else {
-			relativePath = file.Name
-		}
+		var relativePath string = util.GetRelativePath(file)
```
Comment (nishadi) [inline cmd/validate.go]: We can use the := short assignment statement in this var declaration implicitly.

File: util/utils.go [hunk 3]
```diff
@@ -458,7 +459,21 @@ func ProcessString(data, delimiter string, trimAll bool) string {
...
 	if !strings.HasSuffix(archiveFilePath, ".zip") {
-		HandleErrorAndExit(errors.New(fmt.Sprintf("%s must be a zip file. Entered file '%s' is not a valid zip file" +
+		HandleErrorAndExit(errors.New(fmt.Sprintf("%s must be a zip file. Entered file '%s' is not a valid zip file"+
 			".", archiveType, archiveFilePath)))
 	}
 }
+
+// This function will return the relative path of the give file.
+// file	file which needs to obtain relative path
+func GetRelativePath(file *zip.File) (relativePath string) {
```
Comment (nishadi) [inline util/utils.go]: Grammar needs to be corrected. One suggestion is as follows;

- the relative path of the give file --> the relative path of the **_given_** file
- file which needs to obtain relative path --> file in which the relative path is to be obtained

Comment (kasunsiyambalapitiya) [reply util/utils.go]: Fixed in 03f33a8

**Comment by kasunsiyambalapitiya** [comment]
This was closed to avoid a merge commit being pushed and the relevant changes are in new PR #34

**Pull request closed without merging**